source/cipher.c

```diff
@@ -88,6 +88,56 @@ static enum aws_cryptosdk_sha_version aws_cryptosdk_which_sha(enum aws_cryptosdk
     }
 }
 
+static bool str_eq(const char *const str1, const char *const str2) {
+    if ((str1 && !str2) || (!str1 && str2)) {
+        return false;
+    }
+    if (!str1 && !str2) {
+        return true;
+    }
+#ifdef CBMC
+    /* TODO: Bound the alg_props_string size to 128 characters (or any
+       reasonable max size).  This is only useful for CBMC to work,
+       not a real precondition. (We have to include that in the
+       validity check for alg_props (checking for that bound) so that
+       it is at least checked in debug_mode. */
+    /* For now just assert that they both point somewhere */
+    return true;
+#else
+    return strcmp(str1, str2) == 0;
+#endif
+}
+
+static bool alg_properties_equal(
+    const struct aws_cryptosdk_alg_properties alg_props1, const struct aws_cryptosdk_alg_properties alg_props2) {
+    return
+        /* Note: strcmp messes with CBMC so we should either define
+           this to be different for CBMC or bound the size of names */
+        str_eq(alg_props1.md_name, alg_props2.md_name) && str_eq(alg_props1.cipher_name, alg_props2.cipher_name) &&
```

Comment:
Readability:
For readability and consistency, please stick to one check per line.